uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/components/leave-type-wizard.component.js

```diff
@@ -40,42 +44,95 @@ define([
     vm.componentsPath =
       sharedSettings.sourcePath + 'leave-type-wizard/components';
     vm.contacts = null;
+    vm.disableFormSubmission = true;
     vm.fieldsIndexed = {};
-    vm.leaveTypeCategories = [];
+    vm.leaveTypeCategories = null;
     vm.loading = true;
     vm.sections = _.cloneDeep(formSections);
 
     vm.$onInit = $onInit;
+    vm.cancel = cancel;
     vm.checkIfAccordionHeaderClicked = checkIfAccordionHeaderClicked;
     vm.goBack = goBack;
     vm.goNext = goNext;
     vm.openActiveSectionTab = openActiveSectionTab;
     vm.openSection = openSection;
+    vm.submit = submit;
 
     function $onInit () {
       vm.loading = true;
+      vm.isEditMode = !!vm.leaveTypeId;
 
-      loadLeaveTypeCategories()
+      $q.all([
+        loadLeaveTypeCategories(),
+        vm.isEditMode && loadAbsenceType()
+      ])
         .then(initDefaultView)
         .then(function () {
           tabsIndexed = indexPropertyByName(vm.sections, 'tabs');
           vm.fieldsIndexed = indexPropertyByName(tabsIndexed, 'fields');
           promises.titlesInUse = fetchAbsenceTypesTitlesInUse();
+          promises.titlesInUse.then(updateFormSubmissionAllowance);
         })
         .then(markFirstAndLastTabsInSections)
         .then(initDefaultValues)
+        .then(vm.isEditMode && initExistingValues)
         .then(initFieldsWatchers)
         .then(initValidators)
         .then(initCustomValidators)
+        .then(vm.isEditMode && openSettingsSection)
+        .then(vm.isEditMode && adjustCategoryFieldForEditingMode)
         .then(function () {
           vm.loading = false;
         })
+        .then(validateTitleField)
         .then(function () {
           return $q.all([
             loadContacts(),
             loadAvailableColours()
           ]);
+        })
+        .then(vm.isEditMode && assumeAllSectionsAreValid)
+        .then(updateFormSubmissionAllowance);
+    }
+
+    /**
+     * Adjusts Category field in edit mode:
+     * - sets the field to column view by resetting the `labelLayout` property
+     * - updates the label
+     * - makes field not required
+     *
+     * @NOTE it is expected that in UI the user is not able to edit this field
+     */
+    function adjustCategoryFieldForEditingMode () {
+      var field = vm.fieldsIndexed.category;
+
+      delete field.labelLayout;
+      delete field.required;
```

Comment:
is it ok to remove the fields values? I'm guessing that if I want to create a new leave type or modify an existing one it will refresh the page, right?

Reply:
Yes, you cannot jump between modes or absence types without refreshing.